File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/ui/LoginActivity.java
package de.dimedis.mobileentry.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;

import de.dimedis.mobileentry.fragments.SoftwareVersionsFragment;

public class LoginActivity extends BaseFragmentActivity {

    public static void start(@NonNull Context context) {
        Intent intent = new Intent(context, LoginActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (savedInstanceState == null) {
            addFragment(SoftwareVersionsFragment.builder().build(), "init");
        }
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/ui/LoginActivity.kt
package de.dimedis.mobileentry.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import de.dimedis.mobileentry.fragments.SoftwareVersionsFragment
import de.dimedis.mobileentry.util.PrefUtils

class LoginActivity : BaseFragmentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        PrefUtils.setKioskModeEnabled(true)
        if (savedInstanceState == null) {
            addFragment(SoftwareVersionsFragment.builder().build(), "init")
        }
    }

    override fun onBackPressed() {
        // Block Back button in the kiosk mode.
        if (!PrefUtils.isKioskModeEnabled()) {
            super.onBackPressed()
        }
    }

    companion object {
        fun start(context: Context) {
            val intent = Intent(context, LoginActivity::class.java)
            context.startActivity(intent)
        }
    }
}